boards/arm/mps2_an521/dts.fixup

```diff
@@ -0,0 +1,81 @@
+/* SoC level DTS fixup file */
+
+#define CONFIG_NUM_IRQ_PRIO_BITS	ARM_V8M_NVIC_E000E100_ARM_NUM_IRQ_PRIORITY_BITS
```

Comment:
shouldn't most or all of this file go into an soc-level fixup now?

Reply:
Since the MPS2+ ANxyz SoC and board are 1:1, and we have multiple ANxyz we support, I've moved the dts.fixup back into the board for the mps2 only since they have different memory maps and I didn't want to have a huge ifdef block between things.

Reply:
@galak  does this have Security Extension? 

[Will review this PR tomorrow, thoroughly]

Reply:
> @galak does this have Security Extension?

Yes.

Reply:
http://infocenter.arm.com/help/topic/com.arm.doc.dai0521a/DAI0521A_example_castor_subsystem_for_v2m_mps2.pdf